CanvasCore/CanvasCore/React Native Modules/SiriShortcutManager.swift

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright (C) 2018-present Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3 of the License.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import CanvasKit
+import ReactiveSwift
+             
+public class SiriShortcutManager: NSObject {
+
+    enum ShortcutType: String {
+        case grades = "com.instructure.siri.shortcut.getgrades"
+
+        func title(userInfo: [String: Any]) -> String {
+            switch self {
+            case ShortcutType.grades:
+                guard let name = userInfo["name"] as? String else { return "" }
+                let title = NSLocalizedString("Get Grades for", comment: "")
+                return "\(title) \(name)"
+            }
+        }
+    }
+
+    @objc
+    public func donateSiriShortcut(_ userInfo: [String: Any]) {
+        guard let identifier = userInfo["identifier"] as? String,
+              let shortcutType = ShortcutType(rawValue: identifier) else { return }
```

Comment:
This seems like a safer place to check for `userInfo["name"]` instead of in `ShortcutType.title(userInfo:)` because  if it's not there we won't create the activity rather than creating the activity and showing an empty string as the title.